### ISSUE TITLE
background body wont change when toggle dark/light color scheme in next js 13

### ISSUE THREAD

**Issue by anasrar** [mantinedev/mantine#2899]
### What package has an issue

@mantine/core

### Describe the bug

when call `toggleColorScheme` background body wont change in next js 13 but other component does change follow the color scheme

### What version of @mantine/hooks page do you have in package.json?

5.6.2

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/charming-ramanujan-lvqg3n

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

**Comment by rtivital** [comment]
Make sure that you've enabled global styles on MantineProvider – https://mantine.dev/theming/mantine-provider/#css-reset-and-global-styles

**Comment by anasrar** [comment]
> Make sure that you've enabled global styles on MantineProvider – https://mantine.dev/theming/mantine-provider/#css-reset-and-global-styles

already there
https://github.com/anasrar/next-mantine-toggle-dark/blob/9f5fe481ea7da8e1187200c30b90ead0044d7bb9/app/emotion.tsx#L26-L32

test base on next-13-app branch
https://github.com/mantinedev/mantine-next-template/tree/next-13-app

**Comment by anasrar** [comment]
some quick hack for this issue, you can set background and color on `toggleColorScheme`, make sure to add `withCSSVariables` to `MantineProvider`

```js
const [colorScheme, setColorScheme] = useState<ColorScheme>("dark");
const toggleColorScheme = () => {
    setColorScheme(colorScheme === "dark" ? "light" : "dark");
    document.body.style.background =
        colorScheme === "dark"
            ? "var(--mantine-color-white)"
            : "var(--mantine-color-dark-7)";
    document.body.style.color =
        colorScheme === "dark"
            ? "var(--mantine-color-black)"
            : "var(--mantine-color-dark-0)";
};
```
feel free to close this issue

**Comment by dexter4life** [comment]
NextLink showing error message in Nextjs 13 how to fix this problem?

**Comment by rtivital** [comment]
Do not use NextLink

**Comment by joelzwarrington** [comment]
> NextLink showing error message in Nextjs 13 how to fix this problem?

You can use the `legacyBehaviour` prop on NextLink so that it continues to work.

**Comment by thi-marques** [comment]
In the app/**layout.tsx** component the `body` must also be wrapped.

**Comment by CodeMode365** [comment]
Check this out how i fix this is just 1 step:

Before:
**Code inside:** _app/emotion.tsx_
<img width="567" alt="image" src="https://github.com/mantinedev/mantine/assets/102400760/ef9e28de-c73f-4de0-9263-899bdd35a77c">

Result:
<img width="329" alt="image" src="https://github.com/mantinedev/mantine/assets/102400760/b48c75bb-b51e-48cb-823e-cb5fe3b11b2b">

After:
**Code inside:** _app/emotion.tsx_
<img width="554" alt="image" src="https://github.com/mantinedev/mantine/assets/102400760/0ada2d80-f253-4ffd-913f-ed35dfc92dcc">

Result:
<img width="323" alt="image" src="https://github.com/mantinedev/mantine/assets/102400760/8aff6f3e-ae7a-486b-aa1d-52f5420949e1">

**Comment by rtivital** [comment]
This issue is closed for one of these reasons:

- It was fixed in 7.0
- It is no longer reproducible in 7.0
- It is no longer applicable in 7.0
- It does not have a reproduction

If you think that this issue was closed by mistake and it is still an issue in version 7.0, please reopen it.
